### PR TITLE
Add integration tests with Jest

### PR DESCRIPTION
**Pull request by cirex-web**
# Description

I was bored and wanted to refactor the code, but I realized that this codebase was severly lacking in tests so I wrote a single integration test instead. (more to come, probably)

 I was going to use `bun test`, but its mocking is a bit subpar, so I set up Jest (with Babel) instead. All Axios requests are redirected to local HTML files, and all code in `DiningParser` is essentially executed by this one test.

- Also removed the Coordinates class in favor of a simpler `ICoordinate` interface object
- Renamed all interfaces to start with I
- Added a github action for tests
- Updated README

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

**Test Configuration**:

- Node.js version:
- Python version:
- Desktop/Mobile:
- OS:
- Browser:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by cirex-web** [comment]
okay i have no idea why the test github Action isn't running 😭

**Comment by GhostOf0days** [comment]
> okay i have no idea why the test github Action isn't running 😭 (it ran for the first commit)

this has happened to me so many times lol. usually runs if i close and re-open prs.

**Comment by cirex-web** [comment]
nvm I fixed the github action for tests

**Comment by GhostOf0days** [comment]
what was the issue?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Run Tests
+on:
+  push:
+  pull_request:
+    types: [opened, reopened, synchronize]
```
Comment (cirex-web) [inline .github/workflows/test.yml:5]: just had to add synchronize

Comment (cirex-web) [reply .github/workflows/test.yml:5]: https://github.com/orgs/community/discussions/24567

Comment (GhostOf0days) [reply .github/workflows/test.yml:5]: Oh I see. Maybe we also need to add in this to the GitHub Action for CMUEats.

Comment (cirex-web) [reply .github/workflows/test.yml:5]: yes

**Comment by GhostOf0days** [comment]
I think code looks good from first glance. I'll look more into it tomorrow. Small request: can you please update the documentation to include https://bun.sh/docs/cli/update and https://bun.sh/docs/cli/outdated since you also updated ReadMe in this PR? We can also add this to CMUEats after.

**Review by GhostOf0days — CHANGES_REQUESTED** [review]
Thank you for this. Looks good to me. However, I have one small request: can you please update the documentation to include https://bun.sh/docs/cli/update and https://bun.sh/docs/cli/outdated since you also updated the ReadMe in this PR? We can also add this to CMUEats after.

**Comment by cirex-web** [comment]
something like this? idk what you meant exactly lol

## Before submitting a PR

- Make sure all tests pass with `bun run test` or `bun run test --watch` for watch mode.
- Run `bun update` to update all dependencies to the latest version (`bun outdated` to see what will get updated)

**Comment by GhostOf0days** [comment]
Maybe this: Run `bun outdated` to see what dependencies are outdated. Run `bun update` to update all outdated dependencies to the latest version. 

(sorry. i would do this myself, but my laptop's dead, and i'm using github mobile. 😂)

**Review by GhostOf0days — APPROVED** [review]

**Pull request merged**